src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -168,6 +169,30 @@ class DiscreteDistribution
     return log(Probability(observation));
   }
 
+  /**
+   * Calculates the discrete probability density function for each
+   * data point (column) in the given matrix.
+   *
+   * @param x List of observations.
+   * @param probabilities Output probabilities for each input observation.
+   */
+  void Probability(const arma::mat& x, arma::vec& probabilities) const
+  {
+    arma::vec probability(x.n_cols);
+    for (size_t i = 0; i < x.n_cols; i++)
+      probability(i) = Probability(x.unsafe_col(i));
+    probabilities = probability;
+  }
```

Comment:
Do you think we can directly use the output vector here?

Reply:
Ah, nice catch, not sure how I overlooked this. :+1:

Reply:
Yeah, I think we should go by directly using the output vector. I am not sure of memory management of Armadillo. I am confused that how should we define the size of the output variable.

Also let me know if these lines work


```
void Probability(const arma::mat& x, arma::vec& probabilities) const
  {
    for (size_t i = 0; i < x.n_cols; i++)
      probabilities(i) = Probability(x.unsafe_col(i));
  }
```

Reply:
@rcurtin , @ShikharJ I will add the style changes once you help me out with this. Thanks

Reply:
You can set the size of `probabilities` with `probabilities.set_size(x.n_cols);` :+1: